Ou/test/settings quota display check

OliwierUrbanski opened the pull request:
🧪 Quota Display Check (Settings > User)
This PR introduces a complete test for validating the quota usage display inside the HiDrive Next app under Settings → User.

✅ What’s tested:
The correct storage percentage is shown and is between 0–100%, allowing ±1% rounding tolerance.

The usage string follows the expected format:
xx MB of XX GB in use

The used MB value is verified to always be greater than 0.

Output is formatted nicely: 0.04 GB of 1 000 GB used (0 %)

A UI dump (ui_dump.xml) is written to disk for debugging any XPath changes.

OliwierUrbanski committed:
test(settings-quota): add robust quota display E2E check for HiDrive Next

* use pyautogui clicks to bring HiDrive Next to the foreground and open Settings
* start Appium session, select first storage checkbox
* verify quota info in Settings › User:
  - percentage shown and 0-100 % (±1 % tolerance to calc value)
  - “xx MB of XX GB in use” string parsed; total GB matches EXPECTED_TOTAL_GB
  - used MB > 0
* pretty-print used space (GB with two decimals) in console output
* write ui_dump.xml on every run for quick XPath debugging
* keep XPaths and EXPECTED_TOTAL_GB constants at top for easy maintenance

bogus78 commented on TestplanHDNX/test_settings_quota_display_check.py:
```diff
@@ -0,0 +1,150 @@
+    opts = Capabilities.get_options()
+    driver = webdriver.Remote("http://localhost:4723", options=opts)
+    driver.implicitly_wait(8)
+    return driver
```
do we need this initialization and preparation in all test classes? move them to a helper class to avoid duplicate code and use reusable code for better maintainability

bogus78 commented on TestplanHDNX/test_settings_quota_display_check.py:
for what is this commit?

bogus78 commented:
add also a env.example with needed empty variables and a note to the readme

bogus78 commented on TestplanHDNX/test_settings_quota_display_check.py:
```diff
@@ -0,0 +1,150 @@
+    """HiDrive Next in den Vordergrund holen und »Einstellungen« öffnen."""
+    pyautogui.click(3073, 12);  time.sleep(0.3)   # Menüleisten-Icon
+    pyautogui.click(2855, 86);  time.sleep(0.3)   # Benutzer-Dropdown
+    pyautogui.click(2858, 289); time.sleep(0.3)   # «Einstellungen»
```
The coordinates can also be used as constants

bogus78 commented:
proper logging - use a logger instead of print for better monitoring and debugging

bogus78 commented on TestplanHDNX/test_settings_quota_display_check.py:
```diff
@@ -0,0 +1,150 @@
+def click_when_ready(driver, xpath: str, desc: str = "", timeout: int = 20):
+    elem = WebDriverWait(driver, timeout).until(
+        EC.element_to_be_clickable((By.XPATH, xpath))
+    )
```
extract this WebDriverWait into a helper class for reusable code

bogus78 commented:
I would suggest to build a kind of a framework pattern, contains all related methods like click(), wait(), findElement() ... and gui related classes. a test class should be as readable as possible. But this can be done in a separate branch.

OliwierUrbanski committed:
renamed file

OliwierUrbanski committed:
deleted files (pushed into the false branch)

OliwierUrbanski committed:
feat(test-framework): refactor HiDrive Next quota test into modular framework

- migrate from print statements to Python’s `logging` module for structured, leveled output
- extract GUI coordinates into `GuiCoordinates` constants for maintainability
- encapsulate PyAutoGUI interactions in `GuiActions` class to centralize UI operations
- introduce `AppiumUtilities` class with methods `start_session()`, `find_element()` and `click_when_ready()` to reduce boilerplate and improve reusability
- separate XPath locators into `XPaths` class and expected values into module-level constants
- isolate parsing and verification logic into standalone functions (`parse_usage`, `verify_quota_display`)
- improve code readability and test clarity by adopting clear method names, type hints, and class-based organization
- write UI dump to `ui_dump.xml` for optional debugging support

This refactor lays the groundwork for future enhancements, including additional reusable helpers and expanded logging configuration for better monitoring and debugging.

OliwierUrbanski committed:
updated gitiginore file; avoiding unnecessary files

bogus78 reviewed (commented):
please review your commits, some of them do have useful changes and some of them can be squashed / removed.

please move the GUÍCooddinates into a global class as well. I would like to have just on place to change the coordinates:

```
class GuiCoordinates:
    MENU_ICON = (3073, 12)
    USER_DROPDOWN = (2860, 86)
    SETTINGS_ITEM = (2865, 289)
    CLICK_PAUSE = 0.3

```
`WebDriverWait().until() `methods can also be reused, as it can also be included in a global class

OliwierUrbanski committed:
refactor(test_settings_quota_display_check): streamline imports and enhance Appium utilities

OliwierUrbanski committed:
feat(gui_coordinates): add GuiCoordinates class for GUI interaction coordinates

OliwierUrbanski committed:
feat(waits): add Waits class for common WebDriverWait calls

OliwierUrbanski commented:
review feedback applied - ready for review

bogus78 reviewed (approved):
review ok, but please apply the helper classes for the other tests as well

OliwierUrbanski committed:
fix(gui_coordinates): correct file path comment in GuiCoordinates class

OliwierUrbanski committed:
refactor(TestLinkCheck): streamline link verification process and integrate Waits class

OliwierUrbanski committed:
refactor(VersionCheck): improve path handling and streamline GUI interactions with helper classes

OliwierUrbanski commented:
review feedback applied - ready for review @bogus78

OliwierUrbanski committed:
feat(run_all): add script to execute test scripts sequentially with error handling

OliwierUrbanski committed:
feat(TestLinkCheck): enhance link verification process and add .env loading functionality

OliwierUrbanski committed:
fix(VersionCheck): correct import path for Capabilities module
